timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/BatchPingableLeaderResourceTests.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.Random;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.paxos.PaxosValue;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BatchPingableLeaderResourceTests {
+
+    private static final Client CLIENT_1 = Client.of("client1");
+    private static final Client CLIENT_2 = Client.of("client2");
+
+    private static final UUID LEADER_UUID = UUID.randomUUID();
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
```

Comment:
<strong>every time a mock returns a mock a fairy dies</strong>

Reply:
deep stubs is a wonderful exception

Reply:
the alternative is to make this into an interface, make an impl of it, that then returns me the mocks I want, which is horrible